geoh5py/objects/geo_image.py

```diff
@@ -521,61 +586,37 @@ def to_grid2d(
-            rotation_rad = np.arctan2(dxy[1], dxy[0])
-            self._rotation = np.rad2deg(rotation_rad)
-
-        return self._rotation
-
-    @rotation.setter
-    def rotation(self, new_rotation):
-        if self.rotation is not None:
-            if self._vertices is None:
-                raise AttributeError("The image has no vertices")
-
-            # Compute current rotation
-            current_rotation = self.rotation
-            if current_rotation is None:
-                current_rotation = 0
-
-            # Compute rotation angle in degrees
-            rotation_angle_deg = new_rotation - current_rotation
+        if (getattr(self, "_vertices", None) is None) and self.on_file:
+            self._vertices = self.workspace.fetch_array_attribute(self, "vertices")
 
-            # Use the origin vertex (vertices[3]) as the center of rotation
-            center = np.array([self._vertices["x"][3], self._vertices["y"][3]])
+        if self._vertices is None and self.image is not None:
+            if self.tag is not None:
+                self.vertices = self.default_vertices
+                self.georeferencing_from_tiff()
+            else:
+                self.vertices = self.default_vertices
 
-            # Move vertices so that center is at origin
-            vertices_centered_x = self._vertices["x"] - center[0]
-            vertices_centered_y = self._vertices["y"] - center[1]
-            vertices_centered = np.array([vertices_centered_x, vertices_centered_y]).T
+        # todo: change the call from vertices to vertices_xyz in the code
+        if self._vertices is not None:
+            return self._vertices.view("<f8").reshape((-1, 3)).astype(float)
 
-            # Compute rotation matrix using numpy
-            rotation_rad = np.radians(rotation_angle_deg)
-            rotation_matrix = np.array(
-                [
-                    [np.cos(rotation_rad), -np.sin(rotation_rad)],
-                    [np.sin(rotation_rad), np.cos(rotation_rad)],
-                ]
-            )
+        return self._vertices
 
-            # Apply rotation
-            rotated_vertices_centered = vertices_centered @ rotation_matrix
+    @vertices.setter
+    def vertices(self, xyz: np.ndarray | list):
+        if isinstance(xyz, list):
+            xyz = np.asarray(xyz)
 
-            # Move vertices back so that center is at original location
-            self._vertices["x"] = rotated_vertices_centered[:, 0] + center[0]
-            self._vertices["y"] = rotated_vertices_centered[:, 1] + center[1]
+        if not isinstance(xyz, np.ndarray) or xyz.shape != (4, 3):
+            raise ValueError("Input 'vertices' must be a numpy array of shape (4, 3)")
 
-            # Update the stored rotation
-            self._rotation = new_rotation
+        xyz = np.asarray(
+            np.core.records.fromarrays(xyz.T, names="x, y, z", formats="<f8, <f8, <f8")
+        )
+        self._vertices = xyz
```

Comment:
You should probably reset the `dip=None` if the vertices change?

Reply:
Sure, but calling dip setter will set vertices, that will set dip to None.

It's okay, as far as dip can be recomputed, but a bit heavy isn't it?

I'll change it your way for now.

Reply:
Hmm no, you can just do this here

`self._dip = None`

So next time dip is requested, it will have to be recomputed.

Reply:
Is it a big deal to recompute _dip and _rotation every time like I did?
It's not consuming, keep the code clear and avoid this kind of errors.

Reply:
Yep that's fine. Not super efficient, but makes sense since we are actually not storing the dip value on file.

Reply:
it's a 4 points matrix, and storing means running it 2 times anyway.